From: SeongJae Park <sj@kernel.org>
Date: Fri, 10 Oct 2025 12:10:47 -0700
Subject: [PATCH] mm/damon/tests/core-kunit: use KUNIT_EXPECT_PTR_EQ() for
 target->pid

The test code is using KUNIT_EXPECT_EQ() for target->pid expectation,
which causes below warning on W=1 build.

    mm/damon/tests/core-kunit.h: In function 'damon_test_commit_ctx_ensure_target_committed':
    include/kunit/test.h:856:20: warning: initialization of 'long long int' from 'struct pid *

Fix it by using the helper for pointers, KUNIT_EXPECT_PTR_EQ().

Reported-by: kernel test robot <lkp@intel.com>
Closes: https://lore.kernel.org/oe-kbuild-all/202510110139.QxBWTjZO-lkp@intel.com/
Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/tests/core-kunit.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mm/damon/tests/core-kunit.h b/mm/damon/tests/core-kunit.h
index 573b3598ef1e..5c6592968317 100644
--- a/mm/damon/tests/core-kunit.h
+++ b/mm/damon/tests/core-kunit.h
@@ -874,7 +874,7 @@ static void damon_test_commit_ctx_ensure_target_committed(struct kunit *test,
 	for (i = 0; i < damon_test_nr_targets(dst); i++) {
 		dst_target = damon_test_nth_target(dst, i);
 		src_target = damon_test_nth_target(src, i);
-		KUNIT_EXPECT_EQ(test, dst_target->pid, src_target->pid);
+		KUNIT_EXPECT_PTR_EQ(test, dst_target->pid, src_target->pid);
 		KUNIT_EXPECT_EQ(test, dst_target->nr_regions,
 				src_target->nr_regions);
 		damon_test_commit_ctx_ensure_regions_committed(dst_target,
-- 
2.39.5

